### PR TITLE
Media sync service

### LINKED ISSUE #1226: Media Sync Service
A Service that can be configured to send notification to external urls that a new asset was uploaded or delete from sulu.

The notification contains the whole Media Api JSON Object so the external service can download the media store titles or something else.

E.g. For a Video Streaming Server

### PR DESCRIPTION
**Pull request by alexander-schranz**
__Tasks:__

- [ ] test coverage
- [ ] gather feedback for my changes
- [ ] submit changes to the documentation
- [ ] added changelog line

__TODO__:

- [x] Event when a Media is added
- [x] Event when a Media is updated
- [x] Event when Media is deleted
- [x] Service Manager
- [x] Event Listener
- [x] Interface für Service
- [x] External Service sends a notification to external urls that a new asset was uplodated or delete from sulu
- [ ] Tests

Configuration:

``` 
sulu_media:
     external_service:
         video_server:
              add: http://www.example.org/api/add-media
              update: http://www.example.org/api/update-media
              delete: http://www.example.org/api/delete-media
```        

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | TODO
| Fixed tickets | fixes #1226 
| BC Breaks     | yes 
| Doc           | TODO

### CODE REVIEW

File: src/Sulu/Bundle/MediaBundle/Media/Service/ExternalService.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    ) {
+        $this->externalService = $externalService;
+        $this->serializer = $serializer;
+        $this->client = new Client();
+    }
+
+    private function makeRequest($JSONstring, $action)
+    {
+        foreach ($this->externalService as $key => $value) {
+            $request = $this->client->post($value[$action]);
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/MediaBundle/Media/Service/ExternalService.php]: @xandrej use the correct HTTP Method for each
 ```
add:       post
update:  put
delete:   delete
```

File: src/Sulu/Bundle/MediaBundle/Media/Service/ExternalService.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            $request->setBody($JSONstring, 'application/json');
+            $res = $request->send();
+            echo $res->getStatusCode();
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/MediaBundle/Media/Service/ExternalService.php]: remove echo
add error log when statuscode is 4** or 5**

**Comment by wachterjohannes** [comment]
@alexander-schranz docs missing

**Comment by danrot** [comment]
@alexander-schranz What about this feature? Is it finished?

And in general, wouldn't it be nice to solve this in a more generic way, maybe with something called a webhook bundle? Would probably be nice to offer more hooks like this.

**Comment by webmozart** [comment]
Closing due to lack of feedback.

**Pull request closed without merging**